From f660e98ab099226e2664c15006213ec2f8cd6e10 Mon Sep 17 00:00:00 2001
From: rleix <rayx.lei@intel.com>
Date: Wed, 26 Sep 2018 09:29:25 +0800
Subject: [PATCH 2/5] Fix issue that the error toast displayed after renamed a
 file

If the file is not exist when invoke the getFileForDocId, it will
throw a FileNotFoundException. Then the error toast will be displayed.
Move to invoke the getFileForDocId ahead of the rename operation.

Change-Id: I352b48ab70486f67ede6b535ccb1b47607596fcc
Tracked-On:
Signed-off-by: Lei,RayX <rayx.lei@intel.com>
---
 core/java/com/android/internal/content/FileSystemProvider.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/com/android/internal/content/FileSystemProvider.java b/core/java/com/android/internal/content/FileSystemProvider.java
index b591163..1279f35 100644
--- a/core/java/com/android/internal/content/FileSystemProvider.java
+++ b/core/java/com/android/internal/content/FileSystemProvider.java
@@ -237,6 +237,7 @@ public abstract class FileSystemProvider extends DocumentsProvider {
 
         final File before = getFileForDocId(docId);
         final File after = FileUtils.buildUniqueFile(before.getParentFile(), displayName);
+        final File beforeVisibleFile = getFileForDocId(docId, true);
         if (!before.renameTo(after)) {
             throw new IllegalStateException("Failed to rename to " + after);
         }
@@ -245,7 +246,6 @@ public abstract class FileSystemProvider extends DocumentsProvider {
         onDocIdChanged(docId);
         onDocIdChanged(afterDocId);
 
-        final File beforeVisibleFile = getFileForDocId(docId, true);
         final File afterVisibleFile = getFileForDocId(afterDocId, true);
         moveInMediaStore(beforeVisibleFile, afterVisibleFile);
 
-- 
2.7.4

